### PR TITLE
Rankhot dev

### PR DESCRIPTION
**Pull request by Rishoban**
This solution for enhancement Rank-Hot Encoder #199

### CODE REVIEW

**Comment by janmotl** [comment]
Nice. 

Based on travis-ci, it is necessary to add 'RankHotEncoder' into `category_encoders/__init__.py`. 
But even after that, I got a few errors in `test_rankhot` on my computer. I suspect some versions of Pandas may cause troubles...

**Comment by Rishoban** [comment]
I will check that

**Comment by wdm0006** [comment]
@PaulWestenthanner are you interested in taking a look at this PR? It needs to be rebuilt with the newer CI to see if the issues are resolved and have some merge conflicts cleaned up.

File: category_encoders/rankhot.py [hunk 1]
```diff
@@ -0,0 +1,263 @@
...
+
+        X_temp = self.transform(X, override_return_df=True)
+        self.feature_names = list(X_temp.columns)
+
+        if self.drop_invariant:
+            self.drop_cols = []
+            generated_cols = util.get_generated_cols(X, X_temp, self.cols)
+            self.drop_cols = [x for x in generated_cols if X_temp[x].var() <= 10e-5]
+            try:
+                [self.feature_names.remove(x) for x in self.drop_cols]
```
Comment (jona-sassenhagen) [inline category_encoders/rankhot.py:98]:
```suggestion
                for x in self.drop_cols:
                    self.feature_names.remove(x)
```

don't use list comprehensions for side effects.

Comment (jona-sassenhagen) [inline category_encoders/rankhot.py:98]: Alternatively do

```suggestion
                self.feature_names = [name for name in self.feature_names if name not in self.drop_cols]
```

**Comment by PaulWestenthanner** [comment]
Hi @Rishoban 
I'm very sorry for being so late to come back to this PR. I've done a lot of work on it and it now fits nicely into the existing structure (features like return_nan as `handle_missing` or `handle_unknown`). I also made all the tests work and could simplify the code in some places.  
Please have a look, (or anybody else). Feedback is always appreciated

**Pull request merged**